### PR TITLE
Adds add federator with less than m of n votes test

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds add federator with less than m of n votes test

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 1]
```diff
@@ -2319,6 +2319,36 @@ void voteFederationChange_commitEmptyFederation_returnsInsufficientMembersRespon
...
+
+            // Voting with  m of n authorizers to create the pending federation
+            federationSupport.voteFederationChange(tx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            federationSupport.voteFederationChange(tx2, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            ABICallSpec addFederationAbiCallSpec = new ABICallSpec("add", new byte[][]{expectedBtcECKey.getPubKey()});
+
+            // Act
+
+            // Voting less than m of n authorizers
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2342]: `Voting with less`

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 2]
```diff
@@ -2319,6 +2319,36 @@ void voteFederationChange_commitEmptyFederation_returnsInsufficientMembersRespon
 
         }
 
+        @Test
+        void voteFederationChange_addFederatorPublicKeyWithLessThanMofNVotes_returnsPendingFederationSizeZeroResponseCode() {
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2323]: `_returnsPendingFederationSizeZeroResponseCode` this is not an existing response code. It is returning `SUCCESSFUL` one right?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 3]
```diff
@@ -2319,6 +2319,36 @@ void voteFederationChange_commitEmptyFederation_returnsInsufficientMembersRespon
...
+
+            // Arrange
+
+            BtcECKey expectedBtcECKey = new BtcECKey();
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2327]: Let's ose BitcoinTestUtils to create a deterministic key

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 4]
```diff
@@ -2319,6 +2319,36 @@ void voteFederationChange_commitEmptyFederation_returnsInsufficientMembersRespon
...
+
+            Transaction tx = getTransactionFromCaller(FederationChangeCaller.FIRST_AUTHORIZED.getRskAddress());
+            Transaction tx2 = getTransactionFromCaller(FederationChangeCaller.SECOND_AUTHORIZED.getRskAddress());
+
+            ABICallSpec createFederationAbiCallSpec = new ABICallSpec("create", new byte[][]{});
+
+            // Voting with  m of n authorizers to create the pending federation
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 3 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 5]
```diff
@@ -2363,6 +2363,40 @@ void voteFederationChange_commitEmptyFederation_returnsInsufficientMembersRespon
...
+            Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, FederationChangeCaller.FIRST_AUTHORIZED.getRskAddress());
+            Transaction tx2 = TransactionUtils.getTransactionFromCaller(signatureCache, FederationChangeCaller.SECOND_AUTHORIZED.getRskAddress());
+
+            ABICallSpec createFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.CREATE.getKey(), new byte[][]{});
+
+            // Voting with m of n authorizers to create the pending federation
+            federationSupport.voteFederationChange(tx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            federationSupport.voteFederationChange(tx2, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            ABICallSpec addFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.ADD.getKey(), new byte[][]{expectedBtcECKey.getPubKey()});
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2386]: this name is not correct, you are not adding a federation but a member, should be `addFederatorBtcPublicKeyAbiCallSpec`

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 6]
```diff
@@ -2363,6 +2363,40 @@ void voteFederationChange_commitEmptyFederation_returnsInsufficientMembersRespon
 
         }
 
+        @Test
+        void voteFederationChange_addFederatorPublicKeyWithLessThanMofNVotes_returnsSuccessfulResponseCodeAndPendingFederationSizeZero() {
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2367]: nit, but i would say `_returnsSuccessfulResponseCode But PendingFederationSizeZero`

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 5 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 6 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2367]: No need. The test is checking `assertThat(federationSupport.getPendingFederationSize(), is(0));`, so this goes better with the name of the test.

**Review by julia-zack — APPROVED** [review]

**Review by nathanieliov — APPROVED** [review]

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**